### PR TITLE
Implementation of setByAddingObjectsFromArray and tests

### PR DESCRIPTION
**Pull request by RG-J**

### CODE REVIEW

File: tests/unittests/Foundation/NSSetTests.mm [hunk 1]
```diff
@@ -104,4 +104,28 @@
...
+}
+
+TEST(NSSet, setByAddingObjectsFromArrayEmpty) {
+    NSArray* emptyArray = @[];
+    NSSet* set = [NSSet set];
+    NSSet* setFromEmptyArray = [set setByAddingObjectsFromArray:emptyArray];
+    ASSERT_TRUE([setFromEmptyArray count] == 0);
+}
+
+TEST(NSSet, setByAddingObjectsFromArrayDuplicate) {
```
Comment (RG-J) [inline tests/unittests/Foundation/NSSetTests.mm]: adding a test to cover null case

**Comment by rajsesh-msft** [comment]
Thanks @RG-J
